lms/djangoapps/instructor/service/grade_services.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import division
+
+from courseware import courses
+from lms.djangoapps.grades.new.course_grade_factory import CourseGradeFactory
+from lms.djangoapps.instructor.views.reports_helpers import (
+    add_section_info_to_breakdown,
+    calculate_up_to_data_grade,
+    delete_unwanted_keys,
+    generate_by_at,
+    generate_filtered_sections,
+    get_course_subsections,
+    order_by_section_block
+)
+from opaque_keys.edx.keys import CourseKey
+from student.models import CourseEnrollment
+
+
+class GradeServices(object):
+
+    def __init__(self, course_id=None):
+        if course_id is not None:
+            self.course_key = CourseKey.from_string(course_id)
+            self.course = courses.get_course_by_id(self.course_key)
+            self.students = CourseEnrollment.objects.users_enrolled_in(self.course_key)
+
+
+    def generate(self, course_id, task_input):
+        """
+        Public method to generate a grade report, by task_type.
+        """
+        course_string = str(course_id)
+        action_name = task_input['task_type']
+        if action_name == 'section_report':
+            return GradeServices(course_string).by_section(task_input['section_block_id'])
+
+        if action_name == 'enhanced_problem_report':
+            return GradeServices(course_string).enhanced_problem_grade()
+
+
+    def get_grades_by_section(self, section_block_id=None):
+        """
+        Public method to get an object with grades data of the course
+        If section_block_id is set the calculations will made
+        through the all sections till, the selected block_id section.
+
+        Returns:
+            1. section_breakdown: Object list with info about all data subsections.
+            2. section_filtered: Object list like section_breakdown, but wihtout assignment type grades,
+                and dropped subsections.
+            3. section_at_breakdown: Object list by section, with assignment types info in that section.
+            4. grade_breakdown: Dict with grade info keyed by assigment type.
+        """
+        all_grades_info = {}
+        data = []
+        for student in self.students:
+            course_grade_factory = CourseGradeFactory().create(student, self.course)
+            grader_result = course_grade_factory.grader_result
+            grader_by_format = course_grade_factory.graded_subsections_by_format
+            gradeset = add_section_info_to_breakdown(grader_result, grader_by_format)
+            course_policy = course_grade_factory.course_data.course.grading_policy
+            gradeset["username"] = student.username
+            gradeset["fullname"] = student.get_full_name()
+            gradeset = generate_filtered_sections(gradeset)
+            gradeset['section_filtered'] = order_by_section_block(gradeset['section_filtered'], course_grade_factory.course_data)
+            gradeset = generate_by_at(gradeset, course_policy)
```

Comment:
nit: please don't use the abbreviation `at` to represent `assignment_type`.  It's hard to read since it just looks like the word "at".